File: app/src/main/java/org/test/koltintest/MainActivity.java
package org.test.koltintest;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        setSupportActionBar(findViewById(R.id.toolbar));

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(this::onNavigationItemSelected);

        if (savedInstanceState == null) {
            onNavigationItemSelected(navigation.getMenu().findItem(R.id.users));
        }
    }

    private boolean onNavigationItemSelected(@NonNull MenuItem item) {
        Toast.makeText(this, item.getTitle(), Toast.LENGTH_SHORT).show(); // TODO código duplicado

        Fragment newFragment = null;
        switch (item.getItemId()) {
            case R.id.users:
                newFragment = UsersFragment.create();
                break;

            case R.id.shapes:
                newFragment = ShapeFragment.create();
                break;
        }

        if (newFragment != null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragmentContent, newFragment)
                    .commit();
        }
        return true;
    }

}

File: app/src/main/java/org/test/kotlin/demo/MainActivity.kt
package org.test.kotlin.demo

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.test.kotlin.demo.shapes.ShapeFragment
import org.test.kotlin.demo.users.UsersFragment
import org.test.toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        navigation.setOnNavigationItemSelectedListener(this::onNavigationItemSelected)

        if (savedInstanceState == null) {
            onNavigationItemSelected(navigation.menu.findItem(R.id.users))
        }
    }

    private fun onNavigationItemSelected(item: MenuItem): Boolean {
        item.title.toast(this)

        when (item.itemId) {
            R.id.users -> UsersFragment.create()
            R.id.shapes -> ShapeFragment.create()
            else -> null
        }?.let {
            supportFragmentManager.beginTransaction()
                    .replace(R.id.fragmentContent, it)
                    .commit()
        }
        return true
    }

}
